### PR TITLE
Update language server to use Esbonio

### PR DESCRIPTION
**Pull request by dennisameling**
Same as https://github.com/mautic/developer-documentation-new/pull/24

The `reStructuredText` VS Code extension is switching to [Esbonio](https://github.com/swyddfa/esbonio#what-about-the-restructuredtext-extension) since v171.0.0 which has various [benefits](https://github.com/vscode-restructuredtext/vscode-restructuredtext/issues/348). When opening the workspace in Gitpod, a bunch of recommendations are now showing up to migrate settings accordingly.

<a href="https://gitpod.io/#https://github.com/mautic/user-documentation/pull/30"></a>

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -5,15 +5,42 @@
...
+link_text = \"$$link_text\" \n\
+link_url = \"$$link_url\" \n\n\
+link.xref_links.update({link_name: (link_text, link_url)})" \
+  > $(LINKSDIR)/$$file_name".py" \
+
+.PHONY: checklinks
+checklinks:
+	$(SPHINXBUILD) -b linkcheck . $(LINKCHECKDIR)
+	@echo
+	@echo "Check finished. Report is in $(LINKCHECKDIR)."
```
Comment (dennisameling) [inline docs/Makefile:42]: This was missing so the `make checklinks` command couldn't run in CI

**Comment by dennisameling** [comment]
@RCheesley @fakela this is ready for review now - it also fixes the CI pipeline, fixing things like build/link checking 🎉

File: docs/getting-started/how-to-update-mautic.md [hunk 2]
```diff
@@ -43,6 +43,6 @@ If you are sure that you have discovered a bug and you want to report it to deve
 
 [command-line-update]: </home/updating-in-command-line>
 [user-interface-update]: </home/updating-in-the-browser>
-[support-forums]: <https://forum.mautic.org/support>
+[support-forums]: <https://forum.mautic.org/c/support/8>
```
Comment (dennisameling) [inline docs/getting-started/how-to-update-mautic.md:46]: This was an actual broken 404 link 😉

**Review by RCheesley — APPROVED** [review]
Works great locally and on Gitpod, and a bonus fixed URL! Thanks @dennisameling 🎉

**Pull request merged**